backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/IModularContractTypeHandler.java

```diff
@@ -74,4 +82,18 @@ default void cancelLinkedContractsIfAllowed(@NonNull final T model, @NonNull fin
 
 	default void createContractIfRequired(@NonNull final T model)
 	{}
+
+	default boolean isModularContractInProgress(@NonNull final ModularFlatrateTermQuery request)
+	{
+		final List<I_C_Flatrate_Term> modularContracts = streamModularContracts(request)
+				.collect(ImmutableList.toImmutableList());
+
+		return !modularContracts.isEmpty();
+	}
+
+	@NonNull
+	default Stream<I_C_Flatrate_Term> streamModularContracts(@NonNull final ModularFlatrateTermQuery request)
+	{
+		return flatrateBL.streamModularFlatrateTermsByQuery(request);
+	}
```

Comment:
I would move these 2 methods in de.metas.contracts.modular.ModularContractService

Reply:
this was added on interface level, because the query is build from the resolved model from each handler, were applicable